### PR TITLE
[YONK-561]: Optimize CoursesMetricsSocial API

### PR DESCRIPTION
**Pull request by ayesha-baig**
This pull request has been generated against this JIRA ticket, https://openedx.atlassian.net/browse/YONK-561. The code changes are related to performance improvement of CoursesMetricsSocial API.

These changes will make sure that instead of getting all social stats in a course, we need to pick only the data of required users in a course. 

@douglashall could you please review these changes?

### CODE REVIEW

File: api/users.rb [hunk 1]
```diff
@@ -180,7 +185,12 @@ def set_template_result(user_id, user_stats, thread_ids)
     end
 
     # Get the number of threads read by each user.
-    users = User.only([:_id, :read_states]).where("read_states.course_id" => course_id)
+    users_read_states_selector = {"read_states.course_id" => course_id}
+    if user_ids
+        users_read_states_selector[:_id.in] = user_ids
+    end
+
+    users = User.only([:_id, :read_states]).where(users_read_states_selector)
```
Comment (douglashall) [inline api/users.rb]: I think you want to find a way to move the database query on line 198 out of the loop over `users` below.

File: api/users.rb [hunk 2]
```diff
@@ -103,6 +104,10 @@ def _user_social_stats(user_id, params)
       content_selector[:created_at.lte] = end_date
     end
 
+    if user_ids
+        content_selector[:author_id.in] = user_ids
+    end
```
Comment (ayesha-baig) [inline api/users.rb]: content_selector is the data selection criteria, if you see line 133 its where it is used to fetch data from database

**Comment by douglashall** [comment]
@ayesha-baig Can you add a comment to the description of this PR explaining the performance improvements that you expect these changes to have?

File: api/users.rb (hunk 2 above)
Comment (douglashall) [inline api/users.rb]: Got it. Thanks. I didn't have the diff expanded all the way.

**Comment by ayesha-baig** [comment]
@douglashall, updated the comment to elaborate on the changes been made.

**Pull request closed without merging**